api/hardwaremanagement/v1alpha1/hardwaretemplate_types.go

```diff
@@ -26,8 +26,10 @@ type NodePoolData struct {
 	Role string `json:"role"`
 	// +kubebuilder:validation:MinLength=1
 	HwProfile string `json:"hwProfile"`
-	// +kubebuilder:validation:MinLength=1
-	ResourcePoolId string `json:"resourcePoolId"`
+	// +optional
```

Comment:
@irinamihai  Could we add a description to this field that clarifies that this `ResourcePoolId` is unrelated to the `ResourcePoolId` on the O2 interface?   I assume this is the hardware manager vendor's `pool.ResourcePoolId`.

Reply:
Sure, added.